platform/wayland/cog-platform-wl.c

```diff
@@ -1794,12 +2026,12 @@ on_export_shm_buffer (void* data, struct wpe_fdo_shm_exported_buffer* exported_b
     }
 
     buffer->exported_buffer = exported_buffer;
-    shm_buffer_copy_contents (buffer, exported_shm_buffer);
+    shm_buffer_copy_contents(buffer, exported_shm_buffer);
 
-    wl_surface_attach (win_data.wl_surface, buffer->buffer, 0, 0);
-    wl_surface_damage(win_data.wl_surface, 0, 0, INT32_MAX, INT32_MAX);
-    request_frame ();
-    wl_surface_commit (win_data.wl_surface);
+    wl_surface_attach(win_data.shell_context->wl_shell_data.wl_surface, buffer->buffer, 0, 0);
```

Comment:
These should be win_data.wl_surface, right? This wouldn't build if HAVE_SHM_EXPORTED_BUFFER is defined.